internal/core/processing.go

```diff
@@ -115,26 +119,33 @@ func ProcessBookmark(req ProcessRequest) (book model.Bookmark, isFatalErr bool,
 	}
 
 	// Save article image to local disk
-	strID := strconv.Itoa(book.ID)
-	imgPath := fp.Join(req.DataDir, "thumb", strID)
-
-	for _, imageURL := range imageURLs {
-		err = downloadBookImage(imageURL, imgPath)
+	for i, imageURL := range imageURLs {
+		err = DownloadBookImage(imageURL, imgPath)
+		if err != nil {
+			if err.Error() == fmt.Sprintf("%s is not a supported image", imageURL) {
```

Comment:
Instead of this, you can create a custom error and check if its in the chain with `errors.Is`:

```go
var ErrNoSupportedImageType = errors.New("unsupported image type")

// func() error
return ErrUnsupportedImageType

// usage

if err != nil && errors.Is(err, ErrUnsupportedImageType) {
    // ...
}
```